File: src/main/java/ru/sicampus/bootcamp2026/repository/MeetingsRepository.java
package ru.sicampus.bootcamp2026.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import ru.sicampus.bootcamp2026.entity.Invitations;
import ru.sicampus.bootcamp2026.entity.Meetings;
import ru.sicampus.bootcamp2026.entity.Users;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;
import java.util.Set;

@Repository
public interface MeetingsRepository extends JpaRepository<Meetings, Long> {
    Optional<Meetings> findByCreatorIdAndDate(Users id, LocalDateTime date);
}

File: server/src/main/kotlin/com/nikshet/industrialbackend/repository/MeetingsRepository.kt
package com.nikshet.industrialbackend.repository

import com.nikshet.industrialbackend.entity.MeetingEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface MeetingsRepository : JpaRepository<MeetingEntity, UUID> {
    fun findByOrganizerId(organizerId: UUID): List<MeetingEntity>

    fun findByOrganizerIdAndStartTimeBetween(
        organizerId: UUID, startTime: java.time.Instant, startTime2: java.time.Instant
    ): List<MeetingEntity>
}